aicsimageio/readers/czi_reader.py

```diff
@@ -0,0 +1,784 @@
+            The fully constructed and fully delayed image as a Dask Array object.
+        """
+        # Always add the plane dimensions if not present already
+        for dim in REQUIRED_CZI_CHUNK_DIMS:
+            if dim not in self.chunk_by_dims:
+                self.chunk_by_dims.append(dim)
+
+        # Safety measure / "feature"
+        self.chunk_by_dims = [d.upper() for d in self.chunk_by_dims]
+
+        # Construct the delayed dask array
+        dims_shape = CziReader._dims_shape_to_scene_dims_shape(
+            czi.get_dims_shape(),
+            scene_index=self.current_scene_index,
+            consistent=czi.shape_is_consistent,
+        )
+
+        valid_dims = []
+        selected_scene_shape: List[int] = []
+        for dim in selected_scene_dims:
+            if dim in dims_shape.keys():
+                valid_dims.append(dim)
+                if dim == DimensionNames.MosaicTile:
+                    selected_scene_shape.append(
+                        dims_shape[DimensionNames.MosaicTile][1]
+                    )
+                elif dim == DimensionNames.Time:
+                    selected_scene_shape.append(dims_shape[DimensionNames.Time][1])
+                elif dim == DimensionNames.Channel:
+                    selected_scene_shape.append(dims_shape[DimensionNames.Channel][1])
+                elif dim == DimensionNames.SpatialZ:
+                    selected_scene_shape.append(dims_shape[DimensionNames.SpatialZ][1])
+                elif dim == DimensionNames.SpatialY:
+                    selected_scene_shape.append(dims_shape[DimensionNames.SpatialY][1])
+                elif dim == DimensionNames.SpatialX:
+                    selected_scene_shape.append(dims_shape[DimensionNames.SpatialX][1])
+                elif dim == "A":  # sAmples from aicspylibczi3
+                    selected_scene_shape.append(dims_shape["A"][1])
+
+        # Constuct the chunk and non-chunk shapes one dim at a time
+        # We also collect the chunk and non-chunk dimension order so that
+        # we can swap the dimensions after we block out the array
+        non_chunk_dim_order = []
+        non_chunk_shape = []
+        chunk_dim_order = []
+        chunk_shape = []
+        for dim, size in zip(
+            valid_dims, selected_scene_shape
+        ):  # zip(selected_scene_dims, selected_scene_shape):
+            if dim in self.chunk_by_dims:
+                chunk_dim_order.append(dim)
+                chunk_shape.append(size)
+            else:
+                non_chunk_dim_order.append(dim)
+                non_chunk_shape.append(size)
+
+        pixel_dict = {
+            "gray8": np.uint8,
+            "gray16": np.uint16,
+            "gray32": np.uint32,
+            "bgr24": np.uint8,
+            "bgr48": np.uint16,
+        }
+
+        pixel_type = pixel_dict[czi.pixel_type]
+
+        if pixel_type is None:
+            raise TypeError(f"Pixel Type: {czi.pixel_type} not supported!")
```

Comment:
Since you're doing a direct access of `pixel_dict`, this clause will never be reached because the above line will raise a `KeyError`. Suggestion below:
```suggestion
        pixel_type = pixel_dict.get(czi.pixel_type)

        if pixel_type is None:
            raise TypeError(f"Pixel Type: {czi.pixel_type} not supported!")
```